### ISSUE TITLE
Java member autocompletion doesn't work with eclipse-jdt-ls

### ISSUE THREAD

**Issue by kmARC** [prabirshrestha/vim-lsp#1442]
# TL;DR:
Typing after an object's `.` as in after `myList.ad` completion does not show any matches, while the LSP server correctly sends all of them.

# Example
Given the following java file, see the in-line comments. 
```java
public class VimLspTesting
{
	public static int testVariableOne = 1;
	public static int testVariableTwo = 2;
	public static void main( String[] args )
	{
            VimLsp             // <--- would complete to VimLspTesting
            VimLspTesting.test // <---  would NOT complete to testVariableOne / Two
	}
}
```

# Received JSON messages

### Working completion (`VimLsp|` => `VimLspTesting|`)
```
Sun Feb 26 10:35:08 2023:["--->",1,"eclipse-jdt-ls",{"method":"textDocument/completion","on_notification":"---funcref---","params":{"context":{"triggerKind":1},"textDocument":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java"},"position":{"character":18,"line":8}}}]
Sun Feb 26 10:35:08 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:35:08 >> document/didChange","type":3}}}]
Sun Feb 26 10:35:08 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:35:08 >> document/completion","type":3}}}]
Sun Feb 26 10:35:09 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"$/progress","jsonrpc":"2.0","params":{"token":"273c7e5a-22d8-483f-a75d-f64e8663998f","value":{"message":"Validate documents","kind":"begin","title":"Validate documents"}}}}]
Sun Feb 26 10:35:09 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"$/progress","jsonrpc":"2.0","params":{"token":"273c7e5a-22d8-483f-a75d-f64e8663998f","value":{"percentage":0,"message":"Validate documents","kind":"report"}}}}]
Sun Feb 26 10:35:09 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:35:09 Reconciled 1. Took 2 ms","type":3}}}]
Sun Feb 26 10:35:09 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"$/progress","jsonrpc":"2.0","params":{"token":"273c7e5a-22d8-483f-a75d-f64e8663998f","value":{"message":"Validate documents","kind":"end"}}}}]
Sun Feb 26 10:35:09 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:35:09 Completion request completed","type":3}}}]
Sun Feb 26 10:35:09 2023:["<---",1,"eclipse-jdt-ls",{"response":{"id":5,"jsonrpc":"2.0","result":{"isIncomplete":false,"items":[{"label":"VimLspTesting - com.mycompany.app","data":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java","pid":"0","rid":"2","decl_signature":"Lcom.mycompany.app.VimLspTesting;","COMPLETION_EXECUTION_TIME":"45"},"insertTextFormat":1,"command":{"arguments":["2","0"],"title":"","command":"java.completion.onDidSelect"},"sortText":"999999164","kind":7,"filterText":"VimLspTesting","insertText":"VimLspTesting","textEdit":{"range":{"end":{"character":18,"line":8},"start":{"character":12,"line":8}},"newText":"VimLspTesting"},"detail":"com.mycompany.app.VimLspTesting"}]}},"request":{"id":5,"jsonrpc":"2.0","method":"textDocument/completion","params":{"context":{"triggerKind":1},"textDocument":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java"},"position":{"character":18,"line":8}}}}]
```
<details><summary>good-java.json</summary>

```json
{
  "response": {
    "id": 5,
    "jsonrpc": "2.0",
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "VimLspTesting - com.mycompany.app",
          "data": {
            "uri": "file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java",
            "pid": "0",
            "rid": "2",
            "decl_signature": "Lcom.mycompany.app.VimLspTesting;",
            "COMPLETION_EXECUTION_TIME": "45"
          },
          "insertTextFormat": 1,
          "command": {
            "arguments": [
              "2",
              "0"
            ],
            "title": "",
            "command": "java.completion.onDidSelect"
          },
          "sortText": "999999164",
          "kind": 7,
          "filterText": "VimLspTesting",
          "insertText": "VimLspTesting",
          "textEdit": {
            "range": {
              "end": {
                "character": 18,
                "line": 8
              },
              "start": {
                "character": 12,
                "line": 8
              }
            },
            "newText": "VimLspTesting"
          },
          "detail": "com.mycompany.app.VimLspTesting"
        }
      ]
    }
  },
  "request": {
    "id": 5,
    "jsonrpc": "2.0",
    "method": "textDocument/completion",
    "params": {
      "context": {
        "triggerKind": 1
      },
      "textDocument": {
        "uri": "file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java"
      },
      "position": {
        "character": 18,
        "line": 8
      }
    }
  }
}
```

</details>

### Broken completion (`VimLspTesting.test|` => `VimLspTesting.test|` [no completion offered])

```
Sun Feb 26 10:30:53 2023:["--->",1,"eclipse-jdt-ls",{"method":"textDocument/completion","on_notification":"---funcref---","params":{"context":{"triggerKind":1},"textDocument":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java"},"position":{"character":38,"line":8}}}]
Sun Feb 26 10:30:53 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:30:53 >> document/didChange","type":3}}}]
Sun Feb 26 10:30:53 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:30:53 >> document/completion","type":3}}}]
Sun Feb 26 10:30:53 2023:["<---",1,"eclipse-jdt-ls",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"26 Feb 2023, 10:30:53 Completion request completed","type":3}}}]
Sun Feb 26 10:30:53 2023:["<---",1,"eclipse-jdt-ls",{"response":{"id":4,"jsonrpc":"2.0","result":{"isIncomplete":false,"items":[{"label":"testVariableOne : int","data":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java","pid":"0","rid":"1","decl_signature":"Lcom.mycompany.app.VimLspTesting;","COMPLETION_EXECUTION_TIME":"13","name":"testVariableOne"},"textEdit":{"range":{"end":{"character":38,"line":8},"start":{"character":26,"line":8}},"newText":"testVariableOne"},"sortText":"999999178","kind":5,"insertTextFormat":1,"insertText":"testVariableOne","command":{"arguments":["1","0"],"title":"","command":"java.completion.onDidSelect"},"detail":"VimLspTesting.testVariableOne : int"},{"label":"testVariableTwo : int","data":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java","pid":"1","rid":"1","decl_signature":"Lcom.mycompany.app.VimLspTesting;","COMPLETION_EXECUTION_TIME":"13","name":"testVariableTwo"},"textEdit":{"range":{"end":{"character":38,"line":8},"start":{"character":26,"line":8}},"newText":"testVariableTwo"},"sortText":"999999178","kind":5,"insertTextFormat":1,"insertText":"testVariableTwo","command":{"arguments":["1","1"],"title":"","command":"java.completion.onDidSelect"},"detail":"VimLspTesting.testVariableTwo : int"}]}},"request":{"id":4,"jsonrpc":"2.0","method":"textDocument/completion","params":{"context":{"triggerKind":1},"textDocument":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java"},"position":{"character":38,"line":8}}}}]
```

<details><summary>bad-java.json</summary>

```json
{
  "response": {
    "id": 4,
    "jsonrpc": "2.0",
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "testVariableOne : int",
          "data": {
            "uri": "file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java",
            "pid": "0",
            "rid": "1",
            "decl_signature": "Lcom.mycompany.app.VimLspTesting;",
            "COMPLETION_EXECUTION_TIME": "13",
            "name": "testVariableOne"
          },
          "textEdit": {
            "range": {
              "end": {
                "character": 38,
                "line": 8
              },
              "start": {
                "character": 26,
                "line": 8
              }
            },
            "newText": "testVariableOne"
          },
          "sortText": "999999178",
          "kind": 5,
          "insertTextFormat": 1,
          "insertText": "testVariableOne",
          "command": {
            "arguments": [
              "1",
              "0"
            ],
            "title": "",
            "command": "java.completion.onDidSelect"
          },
          "detail": "VimLspTesting.testVariableOne : int"
        },
        {
          "label": "testVariableTwo : int",
          "data": {
            "uri": "file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java",
            "pid": "1",
            "rid": "1",
            "decl_signature": "Lcom.mycompany.app.VimLspTesting;",
            "COMPLETION_EXECUTION_TIME": "13",
            "name": "testVariableTwo"
          },
          "textEdit": {
            "range": {
              "end": {
                "character": 38,
                "line": 8
              },
              "start": {
                "character": 26,
                "line": 8
              }
            },
            "newText": "testVariableTwo"
          },
          "sortText": "999999178",
          "kind": 5,
          "insertTextFormat": 1,
          "insertText": "testVariableTwo",
          "command": {
            "arguments": [
              "1",
              "1"
            ],
            "title": "",
            "command": "java.completion.onDidSelect"
          },
          "detail": "VimLspTesting.testVariableTwo : int"
        }
      ]
    }
  },
  "request": {
    "id": 4,
    "jsonrpc": "2.0",
    "method": "textDocument/completion",
    "params": {
      "context": {
        "triggerKind": 1
      },
      "textDocument": {
        "uri": "file:///Users/<user>/tmp/vim-ale-test-bug/my-app/src/main/java/com/mycompany/app/VimLspTesting.java"
      },
      "position": {
        "character": 38,
        "line": 8
      }
    }
  }
}

```

</details>

# Working Python example

<details><summary>Working python example...</summary>

```python
class VimLspTesting:
    testVariableOne = 1
    testVariableTwo = 2

vim_lsp_testing = VimLspTesting()

VimLspTesting.testVariable # <--- would offer completion to testVariableOne / Two
```

### Working completion (`VimLspTesting.test|` => `VimLspTesting.testVariableOne|`)

```
Sun Feb 26 10:28:51 2023:["--->",1,"pylsp-all",{"method":"textDocument/completion","on_notification":"---funcref---","params":{"context":{"triggerKind":1},"textDocument":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"},"position":{"character":18,"line":6}}}]
Sun Feb 26 10:28:51 2023:["<---",1,"pylsp-all",{"response":{"id":4,"jsonrpc":"2.0","result":{"isIncomplete":false,"items":[{"label":"testVariableOne","data":{"doc_uri":"file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"},"sortText":"atestVariableOne","kind":6,"insertText":"testVariableOne"},{"label":"testVariableTwo","data":{"doc_uri":"file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"},"sortText":"atestVariableTwo","kind":6,"insertText":"testVariableTwo"}]}},"request":{"id":4,"jsonrpc":"2.0","method":"textDocument/completion","params":{"context":{"triggerKind":1},"textDocument":{"uri":"file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"},"position":{"character":18,"line":6}}}}]
```

<details><summary>good-python.json</summary>

```json
{
  "response": {
    "id": 4,
    "jsonrpc": "2.0",
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "testVariableOne",
          "data": {
            "doc_uri": "file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"
          },
          "sortText": "atestVariableOne",
          "kind": 6,
          "insertText": "testVariableOne"
        },
        {
          "label": "testVariableTwo",
          "data": {
            "doc_uri": "file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"
          },
          "sortText": "atestVariableTwo",
          "kind": 6,
          "insertText": "testVariableTwo"
        }
      ]
    }
  },
  "request": {
    "id": 4,
    "jsonrpc": "2.0",
    "method": "textDocument/completion",
    "params": {
      "context": {
        "triggerKind": 1
      },
      "textDocument": {
        "uri": "file:///Users/<user>/tmp/vim-ale-test-bug/vim_lsp_testing.py"
      },
      "position": {
        "character": 18,
        "line": 6
      }
    }
  }
}
```

</details>

</details>

# Environment

### Minimal vimrc
```vim
set nocompatible
filetype plugin indent on
syntax on

call plug#begin('.vim/plugged')
Plug 'prabirshrestha/vim-lsp'
Plug 'mattn/vim-lsp-settings'
call plug#end()

let g:lsp_log_file = expand('/tmp/vim-lsp.log')
let g:lsp_log_verbose = 1

function! s:on_lsp_buffer_enabled() abort
    setlocal omnifunc=lsp#complete
endfunction

augroup lsp_install
    au!
    autocmd User lsp_buffer_enabled call s:on_lsp_buffer_enabled()
augroup END

### Versions

```sh
$ vim --version | head -n3
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Feb 11 2023 11:15:25)
macOS version - arm64
Included patches: 1-1300

$ java --version
openjdk 19.0.2 2023-01-17
OpenJDK Runtime Environment Homebrew (build 19.0.2)
OpenJDK 64-Bit Server VM Homebrew (build 19.0.2, mixed mode, sharing)

[vim-lsp] $  git log -n1
commit 65628c3b0affacd0f36a6e294b1c1f96c6fe2455 (HEAD -> master, origin/master, origin/HEAD)
Date:   Mon Feb 6 14:00:48 2023 +0800

[vim-lsp-settings] $ git log -n1
commit 4492fd36e54c16d4709602e144a55ab5dac9fa34 (grafted, HEAD -> master, origin/master, origin/HEAD)
Date:   Tue Feb 21 13:35:50 2023 +0200
```

**eclipse-jdt-ls**: Downloaded and installed with `:LspUninstallServer` and `:LspInstallServer` today. Current version:
```
 jdt-language-server-latest.tar.gz | 39.9M | 2023-02-23 19:08
```

# Debug steps I tried

1. Tried git bisecting vim-lsp towards `HEAD~32`, this doesn't seem to be working with any versions.
2. Tried disabling `g:lsp_text_edit_enabled`
3. I assumed it might be something with the `refresh_pattern` which is set to `'\k*$` by default. However, both python's and java's `iskeyword=@,48-57,_,192-255`

**Comment by kmARC** [comment]
Hmm indeed it looks like that changing the `refresh_pattern` in `vim-lsp-settings` introduced by [this commit from @mattn](https://github.com/mattn/vim-lsp-settings/commit/270f62e4ad802fe31505d3866e55c8f151c22058) fixes the problem...

From:

```vim
"refresh_pattern": "\\([.a-zA-Z0-9_-]\\+\\|/\\|\\k\\+\\)$"
```

To:
```vim
"refresh_pattern": "\\(/\\|\\k\\+\\)$"
```

Which makes me think if parsing this string is correct? (removing a pattern from the list of accepted patterns seems to solve the problem)
